src/Tools/NodeDocumentationMarkdownGenerator/AssemblyHandler.cs

```diff
@@ -0,0 +1,387 @@
+                
+                foreach (var path in assemblyPaths)
+                {
+                    try
+                    {
+                        var functionDescriptors = new List<FunctionDescriptor>();
+
+                        if (new FileInfo(path).Extension == ".ds")
+                        {
+                            var dsDescriptors = GetFunctionDescriptorsFromDs(pathManager, path);
+                            if (dsDescriptors != null)
+                            {
+                                functionDescriptors.AddRange(dsDescriptors);
+                            }
+                        }
+
+                        else
+                        {
+                            Assembly asm = mlc.LoadFromAssemblyPath(path);
+                            AssemblyName name = asm.GetName();
+
+                            if (NodeModelAssemblyLoader.ContainsNodeModelSubTypeReflectionLoaded(asm, nodeModelType))
+                            {
+                                FileInfosFromNodeModels(asm, nodeModelType, ref nodeSearchModel);
+                                continue;
+                            }
+
+                            var dllDescriptors = GetFunctionDescriptorsFromDll(pathManager, asm);
+                            if (dllDescriptors != null)
+                            {
+                                functionDescriptors.AddRange(dllDescriptors);
+                            }
+                        }                  
+
+                        foreach (var descriptor in functionDescriptors)
+                        {
+                            Dictionary<string, FunctionGroup> fptrs;
+                            if (!functionGroups.TryGetValue(descriptor.Assembly, out fptrs))
+                            {
+                                fptrs = new Dictionary<string, FunctionGroup>();
+                                functionGroups[descriptor.Assembly] = fptrs;
+                            }
+
+                            string qualifiedName = descriptor.QualifiedName;
+                            FunctionGroup functionGroup;
+                            if (!fptrs.TryGetValue(qualifiedName, out functionGroup))
+                            {
+                                functionGroup = new FunctionGroup(qualifiedName);
+                                fptrs[qualifiedName] = functionGroup;
+                            }
+                            functionGroup.AddFunctionDescriptor(descriptor);
+                        }
+                    }
+                    catch (Exception)
+                    {
+                        continue;
+                    }
+                }
+
+                if (functionGroups.Count > 0)
+                {
+                    AddZeroTouchNodesToSearch(functionGroups.Values.SelectMany(x => x.Values.Select(g => g)), ref nodeSearchModel);
```

Comment:
again I don't **think** this should be a ref parameter.

Reply:
No your, not really sure what i was thinking there tbh....